src/NuGet.Core/NuGet.PackageManagement/Utility/PackagesConfigLockFileUtility.cs

```diff
@@ -57,7 +57,7 @@ public static class PackagesConfigLockFileUtility
                     var lockFileRelativePath = projectUri.MakeRelativeUri(lockFileUri).OriginalString;
                     if (Path.DirectorySeparatorChar != '/')
                     {
-                        lockFileRelativePath.Replace('/', Path.DirectorySeparatorChar);
+                        lockFileRelativePath = lockFileRelativePath.Replace('/', Path.DirectorySeparatorChar);
```

Comment:
What's the side-effect here?
There should probably be a test to validate the behavior.

Reply:
I see your description now. 

If there's no functional behavior, then how is this `Replace` even relevant?

Reply:
I haven't actually exercised this code path. I am not sure if the replace is needed or not. Do you think I should just remove this `Replace` call?

Reply:
Looking at this code, it will only be relevant when in Visual Studio (in fact, the code should be tested to see what happens on the command line when this code path is executed, since there won't be a project system, but if it crashes, it's a separate, but related, bug), and possibly only when the lock file does not already exist (and is being opt-in via the msbuild property).

Assigning @jebriede to shepherd this community PR.